controller/appcontroller.go

```diff
@@ -703,7 +703,13 @@ func (ctrl *ApplicationController) normalizeApplication(app *appv1.Application)
 	logCtx := log.WithFields(log.Fields{"application": app.Name})
 	appClient := ctrl.applicationClientset.ArgoprojV1alpha1().Applications(app.Namespace)
 	modifiedApp := app.DeepCopy()
-	modifiedApp.Spec = *argo.NormalizeApplicationSpec(&app.Spec)
+
+	appSourceType, err := argo.QueryAppSourceType(context.Background(), app, ctrl.repoClientset, ctrl.db)
```

Comment:
I think it would be better to return app source type as part of `GenerateManifest`. In this case, type should be available after executing `CompareAppState` in `compareResult` and we don't need to query it again. 

This is just nice to have enhancement. Implement it if you want to.

Reply:
Good idea. Let me give it a try.

Reply:
API sever will error if deploymentInfo.Source.IsZero().